src/realm/object-store/shared_realm.cpp

```diff
@@ -1314,9 +1314,9 @@ SharedRealm Realm::freeze()
     auto config = m_config;
     auto version = read_transaction_version();
     config.scheduler = util::Scheduler::make_frozen(version);
-    config.schema = m_schema;
-    config.schema_version = m_schema_version;
-    return Realm::get_frozen_realm(std::move(config), version);
+    auto frozen_realm = Realm::get_frozen_realm(std::move(config), version);
+    frozen_realm->set_schema(frozen_realm->m_schema, m_schema);
```

Comment:
One concern I have with this fix is that doesn't that break the fixes that was put in for https://github.com/realm/realm-core/issues/4584.

Especially this use case:

> The impetus for this is a case where a mixed column contains a link to a table that doesn't exist in the original user schema (e.g. because it was synced by a later version of the app). If the user attempts to access the mixed property that contains the unknown object, they can get the Core Obj for it, but attempting to then construct an OS Object will fail because [that requires looking up the schema](https://github.com/realm/realm-core/blob/ae1db723d8e757d7c75eb1687a92ab2c65699820/src/realm/object-store/object.cpp#L95-L102) which doesn't exist.

Reply:
No it should not break it at all, essentially we can't longer assign a schema straight and replace the current one. We always need to pass through a bit of logic first, essentially we have to append the mismatching properties at the end of the array of `persisted_properties` and in case of mismatching classes add them and sort things again. 
Otherwise, schemas may diverge, and ugly errors could arise, essentially core is always returning a superset of the schema currently in use.

Reply:
Before to release this, I want to add a bit of testing around this area. We should have caught this.